### ISSUE TITLE
Add site setting to allow anyone to verify all certificates on a site

### ISSUE THREAD

**Issue by marcchristopher** [markn86/moodle-mod_customcert#142]
Hi all,

first of all, thanks for all the hard work! I'm pretty happy with the plugin as it is. However, I seem to be misunderstanding how the link to verify a certificate is supposed to work:

The documentation says: _"Each certificate when issued is assigned a unique code. [...] To validate a certificate a person (eg. employer) may want to enter the code displayed on the certificate to confirm that it is indeed authentic. [...] with this setting set to 'Yes' any person with the verification link (including users not logged in) can do this."_

What is unclear to me: **Which link is that person supposed to click** to be able to enter the code? I do know that there's the menu option 'Verify certificate' in the activity settings menu... but how is someone, who is not logged in, supposed to ever see that link?

In other words: Is there a page that I could link to (say, in the footer), where anyone can enter a code to verify a certificate? I could, of course, simply include that verification link from the activity settings menu... but will I have to do that with every single certificate? That could easily become a very long list. Isn't there a single page for verifying all certificates issued on the site? Like it was possible with that older plugin?

Any help would be much appreciated.
Thanks in advance.

**Comment by markn86** [comment]
Hi there,

Thanks for the report.

The github issues are for bug reports or improvements where as this seems to be a support request. It is better asking these types of questions in the forum https://moodle.org/mod/forum/view.php?id=7163.

However, since you already asked here I will answer. :)

Unfortunately, yes, this will require providing the link to the external user. There is currently no page that the user can go to verify any certificate on the site as the page is certificate dependent, ie. one certificate may allow non-Moodle users to verify and another may not.

A possible solution would be to implement an activity site setting where it would be possible to enable it for all certificates, regardless of their individual settings and then you could provide one link.

Thoughts?

**Comment by marcchristopher** [comment]
Hi,

thanks for the prompt and kind reply.

First of all, sorry for posting this as an issue. Actually, I was implicitly referring to another issue: In [#36](https://github.com/markn86/moodle-mod_customcert/issues/36) you mentioned that the functionality ("a plugin that works like a block verify certificates issued with Custom Certificate") "will exist when [#51](https://github.com/markn86/moodle-mod_customcert/issues/51) is finished". So, I just thought that something must be broken in my installation, as that didn't work.

Anyway, thanks for the suggestion. I think that would be a great idea and would make the plugin even more useful! That new "Custom Certificate" plugin is indeed a huge improvement to the already very useful plugin "Certificate". And there being the option to disallow anyone not logged in to verify a certificate is certainly interesting... but it seems to me that - as it is now - it breaks one of the most critical requirements for making certificates practicable:

Because... to be honest, I fail to see the point of there being a code on a certificate without the possibility to also verify that code. What good is a verification, if only the bearer of a certificate can ever access it? So, at least in my case, I will want to issue quite a few different certificates and never deactivate that setting. After all, it will mostly be employers (or a similar target group) that will want to have easy access to the verification.

But, who knows: There might also be use cases for deactivating verification that I'm not aware of. So, some kind of site setting as you proposed will be a good way of catering for both requirements.

Thanks again for the great work.

**Comment by markn86** [comment]
Hi Marc,

If you set in the certificate instance that you want to be verified by anyone then it will still work as expected. Ie, you create a certificate, set it so anyone can verify, then send the external user (non-Moodle user) the link and they can verify it without ever having to log in. The issue is if you have 1000 certificates on your course then there are going to have to be 1000 individual links you need send out. So, the setting proposed earlier would mean you could send one link and the user could verify any certificate on the site, rather than just one.

**Comment by marcchristopher** [comment]
Hi Mark,

yes, I'm aware of that possibility. I was just referring to the fact that external (non-Moodle) users won't be able to verify a certificate without that specific link. Which is not a realistic use case, in my opinion:

I think, those, who have earned a certificate, will usually want to present it somewhere - say, to a future employer. And maybe that employer will want to verify the validity of the certificate, so the certificate owner will have to send the employer some cryptic link (".../mod/customcert/verify_certificate.php?contextid=..."), say, in an e-mail. That, in my opinion, is not a very elegant option.

It would be better to allow for a simple site-wide verification, just like it was possible with the previous version of the plugin. That way, the employer could simply visit the website, find the global verification link and check the code. So, yes, that setting you proposed would be a great way of handling that.

**Comment by markn86** [comment]
With the proposed change the link will still not be visible to external users as the activity belongs to a course and has limited control (not like a block that can be added to the login page). You would still need to provide a link, albeit it would only be one. Would this still satisfy your requirements?

**Comment by marcchristopher** [comment]
Yes, that'd be perfectly fine. I would then include that link in some central location - say, in the footer of the site - so that guests could easily access it. Will that be possible?

**Comment by markn86** [comment]
Yep, it would.

**Comment by pablojavier** [comment]
Hi Mark! Would this proposed solution work like the "verify certificate" block did, regarding being able to verify any certificate issued by the site just by using the code? I saw in the code that certificates are now more attached to course, and also I saw your comment in code: "It is possible (though unlikely) that there is the same code for issued certificates." That would conspire against a sitewide single point of entry for verifying, wouldn't it?
TIA,
Pablo

**Comment by markn86** [comment]
Hi Pablo,

With this issue resolved there will be one link that someone can visit to verify any certificate on the site. You could simply put this link in the HTML block if you wanted to make it easy to find.

Cheers.

**Comment by markn86** [comment]
Hoping to get this done this weekend and then release a new version on the plugins DB.

**Comment by markn86** [comment]
Will release a new version in the plugins DB this week.
